tanjun/components.py

```diff
@@ -576,7 +628,11 @@ def unbind_client(self, client: abc.Client, /) -> None:
         self._client = None
 
     async def _check_context(self, ctx: abc.Context, /) -> bool:
-        return await utilities.gather_checks(ctx, self._checks)
+        if abc.SlashContext in type(ctx).mro():
```

Comment:
Rather than instance check here, we could just specifically call `utilitiys.gather_checks(ctx, self._slash_checks/self._message_checks)` around line 645 and 656